packages/app/cypress/e2e/integration/top-nav.spec.ts

```diff
@@ -0,0 +1,277 @@
+import type { FoundBrowser } from '@packages/types/src'
+
+describe('App Top Nav Workflows', () => {
+  const setupMockBrowsers = function () {
+    cy.withCtx(async (ctx, o) => {
+      const mockBrowsers = [{
+        channel: 'stable',
+        displayName: 'Chrome',
+        family: 'chromium',
+        name: 'chrome',
+        version: '1.2.333.445',
+        path: '/test/chrome/path',
+        majorVersion: '1',
+      }, {
+        channel: 'stable',
+        displayName: 'Firefox',
+        family: 'firefox',
+        name: 'firefox',
+        path: '/test/firefox/path',
+        version: '2.3.444',
+        majorVersion: '2',
+      }, {
+        channel: 'stable',
+        displayName: 'Electron',
+        family: 'chromium',
+        name: 'electron',
+        path: '/test/electron/path',
+        version: '3.4.555.66',
+        majorVersion: '3',
+      }, {
+        channel: 'stable',
+        displayName: 'Edge',
+        family: 'chromium',
+        name: 'edge',
+        path: '/test/edge/path',
+        version: '4.5.666.77',
+        majorVersion: '4',
+      }] as FoundBrowser[]
+
+      // @ts-ignore sinon is a global in the node process where this is executed
+      sinon.stub(ctx._apis.appApi, 'getBrowsers').resolves(mockBrowsers)
+    })
+  }
+
+  beforeEach(() => {
+    cy.scaffoldProject('launchpad')
+
+    cy.clock(Date.UTC(2021, 10, 31), ['Date'])
+  })
+
+  describe('Page Name', () => {
+    it('shows the current page name in the top nav', () => {
+      setupMockBrowsers()
+      cy.openProject('launchpad')
+      cy.startAppServer()
+      cy.visitApp()
+
+      cy.findByTestId('app-header-bar').should('be.visible').and('contain', 'Specs-Index')
+    })
+  })
+
+  describe('Browser List', () => {
+    context('with command line args', () => {
+      it('shows current browser when launched with browser option', () => {
+        setupMockBrowsers()
+        cy.openProject('launchpad', ['--browser', 'firefox'])
+        cy.startAppServer()
+        cy.visitApp()
+
+        cy.findByTestId('top-nav-active-browser-icon')
+        .should('have.attr', 'src')
+        .and('contain', 'firefox')
+
+        cy.findByTestId('top-nav-active-browser').should('contain', 'Firefox v2')
+      })
```

Comment:
Since we have access to the `mockBrowsers` provided above, I wonder if we can do something like below. It would help cover all browsers in the list and increase the blast radius of covered cases without having to add a whole lot of code. I don't know if we have this pattern in any of our tests or if we want to introduce it, but it looks like we can leverage the same pattern below as well for some of the other tests below. Let me know what you think!

```suggestion
  mockBrowsers.forEach(browser => {
    it(`shows current browser (${browser.name}) when launched with browser option`, () => {
          setupMockBrowsers()
          cy.openProject('launchpad', ['--browser', browser.name])
          cy.startAppServer()
          cy.visitApp()
  
          cy.findByTestId('top-nav-active-browser-icon')
          .should('have.attr', 'src')
          .and('contain', browser.name)
  
          cy.findByTestId('top-nav-active-browser').should('contain', `${browser.displayName} v${browser.majorVersion}`)
      })
  })
```

Reply:
I think this is interesting. The "plumbing" under test here should be the same in each loop, so I'm not sure if we're increasing coverage all that much. At the same time, given we have a pretty controlled group of inputs here, additional checks in a loop like this could give us some rock-solid guarantees. I'll play around with this and see what shakes out.